### PR TITLE
E1670. Unit tests for answers.rb

### PR DESCRIPTION
**Pull request by anudeepvrm**

### CODE REVIEW

File: spec/models/answer_spec.rb [hunk 1]
```diff
@@ -0,0 +1,165 @@
+require 'rails_helper'
+
+describe Answer do
+  let(:questionnaire) {create(:questionnaire,:id => 1)}
```
Comment (Winbobob) [inline spec/models/answer_spec.rb]: It is overkilled to use `create` in unit tests, using `double` or `build` instead.

File: spec/models/answer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  let!(:answer) { create(:answer,:question => question1,:response_id => 1)}
+  
+  describe "# test dependancy between question.rb and answer.rb" 
+    it { should belong_to(:question) }
+  
+  describe "#test get total score" do
+   				
+    it "returns total score when required conditions are met" do
+      # stub for ScoreView.find_by_sql to revent prevent unit testing sql db queries
+      allow(ScoreView).to receive(:find_by_sql).and_return([double("scoreview",weighted_score: 20,sum_of_weights: 5,q1_max_question_score: 4)])
```
Comment (Winbobob) [inline spec/models/answer_spec.rb]: Duplicated code, it will be better to extract a method to deal with.

Comment (anudeepvrm) [inline spec/models/answer_spec.rb]: do I make these changes now or should I wait for the build to get stable ? It is currently failing

File: spec/models/answer_spec.rb (hunk 1 above)
Comment (anudeepvrm) [inline spec/models/answer_spec.rb]: Yeah I have used stubs and doubles everywhere except for testing sql queries. I know it is a bad practice to create test record and test queries but the prof told me to write general tests to see if db schema is up to date. Hence I had to use create for those sql tests.

Comment (Winbobob) [inline spec/models/answer_spec.rb]: OK

**Pull request merged**